### PR TITLE
🌱 Bump to k/k 1.30 RC2, downgrade prometheus, zap, golang.org/x/sys, sigs.k8s.io/yaml

### PR DESCRIPTION
**Pull request by sbueringer**
to k/k versions

Signed-off-by: Stefan Büringer buringerst@vmware.com

Context: we had some problems with compile issues with prometheus/client_.* caused by diverging versions between CR & k/k

#2774 will introduce a tool that ensure the dependencies are always sync. The current PR downgrades the dependencies that are currently out of sync.

### CODE REVIEW

**Comment by sbueringer** [comment]
I'll take a look at the finding in a bit

File: go.mod [hunk 1]
```diff
@@ -12,12 +12,12 @@ require (
...
-	github.com/prometheus/client_golang v1.17.0
-	github.com/prometheus/client_model v0.4.1-0.20230718164431-9a2bf3000d16
+	github.com/prometheus/client_golang v1.16.0
+	github.com/prometheus/client_model v0.4.0
 	go.uber.org/goleak v1.3.0
-	go.uber.org/zap v1.27.0
+	go.uber.org/zap v1.26.0
 	golang.org/x/exp v0.0.0-20220722155223-a9213eeb770e
-	golang.org/x/sys v0.19.0
+	golang.org/x/sys v0.18.0
```
Comment (vincepri) [inline go.mod:20]: Should this be 0.17?

Comment (sbueringer) [reply go.mod:20]: You're right for RC1, I'll bump to RC2 instead

Comment (sbueringer) [reply go.mod:20]: Nice catch!

**Comment by sbueringer** [comment]
@vincepri PTAL

**Review by vincepri — APPROVED** [review]

**Pull request merged**